slim_trees/pickling.py

```diff
@@ -129,29 +135,39 @@ def load_compressed(
                         set to the compression method and other key-value pairs which are forwarded
                         to open() of the compression library.
                         Inspired by the pandas.to_csv interface.
+    :param unpickler_class: custom unpickler class derived from pickle.Unpickler.
```

Comment:
Hm, I wonder if we should pass a callable `load(file) -> unpickled_obj` here instead of a class, and users may pass `load=lambda file: pickle.Unpickler(file).load()`.

Reply:
As shown in the tests, a custom unpickling scheme will be defined via a class derived from `pickle.Unpickler`. Hence, I find it the most straightforward to pass this class directly, rather than taking the function detour.

But I'm open to other solutions as well. What do you think, @pavelzw?

Reply:
I think since we expect users to always provide some kind of `Unpickler`, this approach is fine.
Otherwise, everytime you want to specify custom pickling behavior, you would need to explicitly pass a lambda function `lambda file: CustomUnpickler(file).load()`.